### PR TITLE
Use OVM_GasPriceOracle for Optimism tx fee estimation

### PR DESCRIPTION
**Pull request by miguelmota**

### CODE REVIEW

File: packages/hop-node/src/watchers/classes/Bridge.ts [hunk 1]
```diff
@@ -752,7 +755,24 @@ export default class Bridge extends ContractBase {
...
+        const ovmGasPriceOracle = getContractFactory('OVM_GasPriceOracle')
+          .attach(predeploys.OVM_GasPriceOracle).connect(getRpcProvider(this.chainSlug))
+        const serializedTx = serializeTransaction({
+          value: parseEther('0'),
+          gasPrice,
+          gasLimit,
+          to,
+          data
+        })
+        gasCost = await ovmGasPriceOracle.getL1Fee(serializedTx)
```
Comment (shanefontaine) [inline packages/hop-node/src/watchers/classes/Bridge.ts:771]: I think this value should be added to the existing `gasCost`. IIUC, this value provides the L1 cost, in Wei and needs to be added to our existing calc, `gasCost = gasLimit.mul(gasPrice)`, which is the L2 cost.

File: packages/sdk/src/HopBridge.ts [hunk 2]
```diff
@@ -665,13 +666,21 @@ class HopBridge extends Base {
...
+        const ovmGasPriceOracle = getContractFactory('OVM_GasPriceOracle')
+          .attach(predeploys.OVM_GasPriceOracle).connect(destinationChain.provider)
+        const serializedTx = serializeTransaction({
+          value: parseEther('0'),
+          gasPrice,
+          gasLimit: bondTransferGasLimit,
+          to,
+          data
+        })
+        txFeeEth = await ovmGasPriceOracle.getL1Fee(serializedTx)
```
Comment (shanefontaine) [inline packages/sdk/src/HopBridge.ts:680]: Same comment as hop-node.

**Review by shanefontaine — COMMENTED** [review]
@miguelmota back to you

File: packages/sdk/src/HopBridge.ts (hunk 2 above)
Comment (miguelmota) [reply packages/sdk/src/HopBridge.ts]: Fixed with https://github.com/hop-protocol/hop/pull/287/commits/5f60e11bb197451ff9f6d15ea3edc644dab3f5b0

File: packages/hop-node/src/watchers/classes/Bridge.ts (hunk 1 above)
Comment (miguelmota) [reply packages/hop-node/src/watchers/classes/Bridge.ts]: Fixed with https://github.com/hop-protocol/hop/pull/287/commits/5f60e11bb197451ff9f6d15ea3edc644dab3f5b0

**Pull request merged**